asset/src/kafka_sender_api/kafka-route-sender.ts

```diff
@@ -0,0 +1,113 @@
+import {
+    RouteSenderAPI, AnyObject, DataEntity, getValidDate, isString, Logger, toString
+} from '@terascope/job-components';
+import * as kafka from 'node-rdkafka';
+import { ProducerClient, ProduceMessage } from '../_kafka_clients';
+
+export default class KafkaSender implements RouteSenderAPI {
+    logger: Logger;
+    producer: ProducerClient;
+    readonly hasConnected = false;
+    readonly config: AnyObject = {};
+    readonly isWildcard: boolean;
+    private tryFn: (msg: any, err: any) => DataEntity|null;
+
+    constructor(client: kafka.Producer, config: AnyObject) {
+        const producer = new ProducerClient(client, {
+            logger: config.logger,
+            topic: config.topicOverride || config.topic,
+            bufferSize: config.bufferSize,
+        });
+
+        this.config = config;
+        this.isWildcard = config._key && config._key === '**';
+        this.producer = producer;
+        this.tryFn = config.tryFn || this.tryCatch;
+        this.logger = config.logger;
+    }
+
+    private tryCatch(fn: any) {
+        return (input: any) => {
+            try {
+                return fn(input);
+            } catch (err) {
+                this.logger.warn(`Error computing ${toString(input)}, error: ${err.message}`);
+                return null;
+            }
+        };
+    }
+
+    async initialize(): Promise<void> {
+        await this.producer.connect();
+    }
+
+    async disconnect(): Promise<void> {
+        await this.producer.disconnect();
+    }
+
+    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+    async send(data: DataEntity[]): Promise<void> {
+        const mapper = this.mapFn.bind(this);
+        const resutls = await this.producer.produce(data, mapper);
```

Comment:
Why does this return `void` but return results? Also you misspelled results (`resutls`)